src/graph/parser.ts

```diff
@@ -1,24 +1,30 @@
 import { ucast, calculateHash, Hash } from "../utils";
-import { GraphSettings, Equation, Color, ColorConstant, LineStyle, PointStyle, DegreeMode } from "./interface";
+import { GraphSettings, Equation, Color, ColorConstant, LineStyle, PointStyle, DegreeMode, CSSUnit, Size } from "./interface";
+
 
 /** The maximum dimensions of a graph */
 const MAX_SIZE = 99999;
 
 const DEFAULT_GRAPH_SETTINGS: GraphSettings = {
-    width: 600,
-    height: 400,
+    width: {value: 600, unit: CSSUnit.px},
+    height: {value: 400, unit: CSSUnit.px},
     left: -10,
     right: 10,
     bottom: -7,
     top: 7,
     grid: true,
+    skipCache: false,
     degreeMode: DegreeMode.Radians,
 };
 
 const DEFAULT_GRAPH_WIDTH = Math.abs(DEFAULT_GRAPH_SETTINGS.left) + Math.abs(DEFAULT_GRAPH_SETTINGS.right);
 
 const DEFAULT_GRAPH_HEIGHT = Math.abs(DEFAULT_GRAPH_SETTINGS.bottom) + Math.abs(DEFAULT_GRAPH_SETTINGS.top);
 
+export const RelativeCSSUnit = [
+    CSSUnit.em, CSSUnit.ch, CSSUnit.rem, CSSUnit.vw, CSSUnit.vh, CSSUnit.vmin, CSSUnit.vmax, CSSUnit.percent
+]
+
```

Comment:
You should just have a separate enum for relative and non relative units then store them as a union as use `parseStringToEnum`

Reply:
Fixed: 4b7f10b881b4ebdfd092715b4b2897dae3cd5dc0